File: opencv/Task1_basic_operates/Code/cutout.py
import cv2

# 读取图像文件，替换为你实际的图像文件路径
image_path = "all.jpg"
image = cv2.imread(image_path)
if image is None:
    print("无法读取图像文件，请检查文件路径是否正确")
    exit()

# 确定手机部分在图像中的坐标范围（需根据实际情况调整）
x = 5
y = 105
w = 616
h = 290

# 截取手机部分图像
phone_image = image[y:y + h, x:x + w]

# 显示截取后的手机图像
cv2.imshow('Phone Image', phone_image)
cv2.waitKey(0)
cv2.destroyAllWindows()

# 保存截取后的手机图像（可选步骤，替换为你想要保存的文件名和路径）
save_path = "phone.jpg"
cv2.imwrite(save_path, phone_image)
print(f"截取的手机图像已成功保存至 {save_path}")

File: EmbemdedLinux/Task4_OpenCV_Cpp/Code/cutout.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

int main() {
    // 读取 all.jpg 图像
    cv::Mat image = cv::imread("/root/Desktop/ImageShow/my_venv/photograph/all.jpg");
    if (image.empty()) {
        std::cerr << "Could not open or find the image: all.jpg" << std::endl;
        return -1;
    }

    // 假设手机部分的坐标和尺寸（根据实际情况修改）
    int x = 5;  // 截取区域左上角 x 坐标
    int y = 105;  // 截取区域左上角 y 坐标
    int width = 616;  // 截取区域的宽度
    int height = 290; // 截取区域的高度
    cv::Rect roi(x, y, width, height);

    // 截取手机部分
    cv::Mat phone = image(roi);

    // 保存截取的图像为 phone.jpg
    if (!cv::imwrite("phone.jpg", phone)) {
        std::cerr << "Could not save the image: phone.jpg" << std::endl;
        return -1;
    }

    std::cout << "Image saved successfully as phone.jpg" << std::endl;
    return 0;
}
